File: back/src/main/java/com/triptrace/domain/post/post/entity/Post.java
package com.triptrace.domain.post.post.entity;

import com.triptrace.domain.trip.trip.entity.Trip;
import com.triptrace.global.jpa.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import java.time.LocalDate;

@Entity
public class Post extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trip_id", nullable = false)
    private Trip trip;

    @Column(nullable = false)
    private LocalDate date;

    @Column(length = 100, nullable = false)
    private String title;

    @Lob // TEXT로 매핑
    private String memo;

    public Post(Trip trip, LocalDate date, String title, String memo) {
        this.trip = trip;
        this.date = date;
        this.title = title;
        this.memo = memo;
    }

    public void modify(LocalDate date, String title, String memo) {
        this.date = date;
        this.title = title;
        this.memo = memo;
    }

    public Trip getTrip() {
        return this.trip;
    }

    public LocalDate getDate() {
        return this.date;
    }

    public String getTitle() {
        return this.title;
    }

    public String getMemo() {
        return this.memo;
    }

    public Post() {
    }
}

File: back/src/main/java/com/triptrace/domain/post/post/entity/Post.kt
package com.triptrace.domain.post.post.entity

import com.triptrace.domain.trip.trip.entity.Trip
import com.triptrace.global.jpa.entity.BaseEntity
import jakarta.persistence.*
import java.time.LocalDate

@Entity
class Post(
    trip: Trip,
    date: LocalDate,
    title: String,
    memo: String?
) : BaseEntity() {
    @field:ManyToOne(fetch = FetchType.LAZY)
    @field:JoinColumn(name = "trip_id", nullable = false)
    var trip: Trip = trip
        protected set

    @field:Column(nullable = false)
    var date: LocalDate = date
        protected set

    @field:Column(length = 100, nullable = false)
    var title: String = title
        protected set

    @field:Lob // TEXT로 매핑
    var memo: String? = memo
        protected set

    fun modify(date: LocalDate, title: String, memo: String?) {
        this.date = date
        this.title = title
        this.memo = memo
    }
}
